### ISSUE TITLE
Predefined shapes for Path2D

### ISSUE THREAD

**Issue by nathanjwtx** [godotengine/godot-proposals#1925]
### Describe the project you are working on
Currently working on 2 different types of space shooter where the enemies travel along paths. 
### Describe the problem or limitation you are having in your project
Currently there is no facility to create predefined shapes such as you can for Collision2D for example. This means that trying to create a **smooth** circular path is very difficult.
### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Add some predefined shapes for creating curved paths. Or enhance existing tooling to make creating curves easier.
### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
![image](https://user-images.githubusercontent.com/1558027/100895079-c9257480-3482-11eb-9e00-19f6b455d989.png)
Replicate this functionality to provide circle, capsule shapes etc. This is less of an issue for non-curved shapes/surfaces.
### If this enhancement will not be used often, can it be worked around with a few lines of script?
I guess it could be done with a script but it would be time consuming and complicated to calculate the positions of the points required for a circle.
### Is there a reason why this should be core and not an add-on in the asset library?
It would bring Path2D into alignment with other node types that offer predetermined shapes.

**Comment by Calinou** [comment]
> I guess it could be done with a script but it would be time consuming and complicated to calculate the positions of the points required for a circle.

`Vector2.rotated()` should make this quite easy :slightly_smiling_face: 

```gdscript
tool
extends Path2D


func _ready() -> void:
	curve = Curve2D.new()
	for i in 32:
		curve.add_point(Vector2(0, -100).rotated((i / 32.0) * TAU))

	# End the circle.
	curve.add_point(Vector2(0, -100))
```

Results in:

![image](https://user-images.githubusercontent.com/180032/100907151-40cac200-34ca-11eb-9044-8bb68a2703eb.png)

**Comment by nathanjwtx** [comment]
Thanks! Let me give that a try. :)

**Comment by Xrayez** [comment]
I think what would be useful is creating a general-purpose polygon generator, including (and not limited) to those `Shape2D` resources which were rather meant to be used as collision shapes.

See **Flattening/converting collision shapes into polygon-based shapes** topic at godot-extended-libraries/godot-ideas#15 for related GIP proposals.

**Comment by nathanjwtx** [comment]
I agree. That would seem to address lots of similar issues where predefined shapes would simplify the task in hand.

**Comment by chucklepie** [comment]
I'm now moving into path2d for some of my more complex actions and want symmetrical and perfectly curved irregular shapes. I just gave up. I hope this suggestion gets some traction...

**Comment by skeddles** [comment]
yes please, I dont wanna have to hand draw a circle or find this thread so i can copypaste that script every time
